dali/imgcodec/parsers/webp.cc

```diff
@@ -12,19 +13,155 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
 #include "dali/imgcodec/parsers/webp.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+namespace {
+
+// Chunk specification:
+// https://developers.google.com/speed/webp/docs/riff_container#riff_file_format
+//
+// Header layout:
+// RiffHeader | vp8_header_t | <WebpLossyHeader/WebpLosslessHeader>
+//
+// The WebpLossyHeader and WebpLosslessHeader are selected based on a identifier,
+// respectively "VP8 " and "VP8L" in vp8_header_t. "VP8X" in vp8_header_t signifies
+// WebP extended file format, which is not supported.
+
+struct RiffHeader {
+  std::array<uint8_t, 4> riff_text;
+  std::array<uint8_t, 4> file_size;
+  std::array<uint8_t, 4> webp_text;
+};
+static_assert(sizeof(RiffHeader) == 12);
+
+using vp8_header_t = std::array<uint8_t, 4>;
+
+// Simple file format (lossy)
+// https://datatracker.ietf.org/doc/html/rfc6386#section-9.1
+struct WebpLossyHeader {
+  std::array<uint8_t, 4> chunk_size;
```

Comment:
```suggestion
#pragma pack(push, 1)
struct WebpLossyHeader {
  uint32_t chunk_size;
```

and adjust endianness (if needed) upon reading?